xds/internal/xdsclient/pool.go

```diff
@@ -0,0 +1,143 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xdsclient
+
+import (
+	"fmt"
+	"sync"
+
+	"google.golang.org/grpc/internal/backoff"
+	"google.golang.org/grpc/internal/grpcsync"
+	"google.golang.org/grpc/internal/xds/bootstrap"
+)
+
+var (
+	// DefaultPool is the default pool for xds clients. It is created at the
+	// init time.
+	DefaultPool *Pool
+)
+
+// Pool represents pool of xds clients.
+type Pool struct {
+	mu      sync.Mutex
+	clients map[string]*clientRefCounted
+	config  *bootstrap.Config
```

Comment:
Nit: `mu` should ideally *only* have to guard `clients`. But here, we need it to guard `config` as well since `SetFallbackBootstrapConfig` writes to `config`. Maybe a comment here to mention that?

Reply:
Added in the docstring